include/pybind11/stl_bind.h

```diff
@@ -559,7 +562,7 @@ template <typename Map, typename Class_> auto map_if_insertion_operator(Class_ &
             for (auto const &kv : m) {
                 if (f)
                     s << ", ";
-                s << kv.first << ": " << kv.second;
+                s << to_string(kv.first) << ": " << to_string(kv.second);
```

Comment:
Hm, going to Python here strikes me really as overkill. We should rather expect the user to provide an `operator<<` if they want to be able to stringify their objects. Can you please remove this part of the change?

Reply:
agreed